pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTArrayLoadExpression.java

```diff
@@ -7,12 +7,14 @@
 import com.google.summit.ast.Node;
 import net.sourceforge.pmd.annotation.InternalApi;
 
-public class ASTArrayLoadExpression extends AbstractApexNode.Single<Node> {
+import com.google.summit.ast.expression.ArrayExpression;
+
+public class ASTArrayLoadExpression extends AbstractApexNode.Single<ArrayExpression> {
 
     @Deprecated
     @InternalApi
-    public ASTArrayLoadExpression(Node arrayLoadExpression) {
-        super(arrayLoadExpression);
+    public ASTArrayLoadExpression(ArrayExpression arrayExpression) {
+        super(arrayExpression);
     }
```

Comment:
Just FYI: The constructor is deprecated+internal API because it should have been actually package private - only be called by the tree builder.
Since we are changing here the API anyway, anyone who currently creates manually AST nodes would need to adapt their code when switching to summit.
Maybe we can clean this up at the end and make this constructors package private (a task for the doc)? I'm not entirely clear about the consequences yet...

Reply:
Thanks, makes sense. I went ahead and made all of the new ctors package-private and removed the annotations. I recorded the documentation requirement in the follow-up tasks.